### PR TITLE
Added context manager for recording in python

### PR DESCRIPTION
**Pull request by bsaund**
@LemonPi I enjoyed the context manager in your rviz screen recorder, and made something similar for the video camera recorder

### CODE REVIEW

File: src/arm_video_recorder/camera_recorder.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import rospy
+from arm_video_recorder.srv import TriggerVideoRecording, TriggerVideoRecordingRequest
+from datetime import datetime
+from pathlib import Path
+
+
+class CameraRecorder:
+
+    def __init__(self, filename=None, timestamp_format='%Y_%m_%d_%H_%M_%S', timeout_seconds=600.0):
```
Comment (LemonPi) [inline src/arm_video_recorder/camera_recorder.py]: Having filename as optional but defaulting to None is questionable. Does Path(None) default to home directory? If so, please document or make filename a required argument

File: src/arm_video_recorder/camera_recorder.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        self.save_file = filename
+        self.timestamp_format = timestamp_format
+        self.video_srv = rospy.ServiceProxy("/video_recorder", TriggerVideoRecording)
+        self.timeout = timeout_seconds
+
+    def __enter__(self):
+        req = TriggerVideoRecordingRequest()
+        f = Path(self.save_file)
```
Comment (LemonPi) [inline src/arm_video_recorder/camera_recorder.py:17]: Maybe add a logging here for where it's saved to?

File: src/arm_video_recorder/camera_recorder.py (hunk 1 above)
Comment (LemonPi) [reply src/arm_video_recorder/camera_recorder.py]: alternatively can leave as is if you log where it's saved to in the second comment

Comment (bsaund) [reply src/arm_video_recorder/camera_recorder.py]: Agreed, it is unclear what `None` does. Removed default

**Pull request merged**